x/psiphon/psiphon.go

```diff
@@ -76,162 +77,111 @@ var _ transport.StreamDialer = (*Dialer)(nil)
 // you will need to add it independently.
 func (d *Dialer) DialStream(unusedContext context.Context, addr string) (transport.StreamConn, error) {
 	d.mu.Lock()
-	controller := d.controller
+	tunnel := d.tunnel
 	d.mu.Unlock()
-	if controller == nil {
+	if tunnel == nil {
 		return nil, errNotStartedDial
 	}
-	netConn, err := controller.Dial(addr, nil)
+	netConn, err := tunnel.Dial(addr)
 	if err != nil {
 		return nil, err
 	}
 	return streamConn{netConn}, nil
 }
 
-func newPsiphonConfig(config *DialerConfig) (*psi.Config, error) {
+func getClientPlatform() string {
+	clientPlatformAllowChars := func(r rune) bool {
+		return !unicode.IsSpace(r) && r != '_'
+	}
+	goos := strings.Join(strings.FieldsFunc(runtime.GOOS, clientPlatformAllowChars), "-")
+	goarch := strings.Join(strings.FieldsFunc(runtime.GOARCH, clientPlatformAllowChars), "-")
```

Comment:
Do we really need to normalize these two strings? It seems all [`GOOS`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/internal/goos/zgoos_linux.go;l=7) and [`GOARCH`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/internal/goarch/zgoarch_amd64.go;l=7) are already normalized.

Reply:
I copied the code from @adam-p in https://github.com/Jigsaw-Code/outline-sdk/pull/251.

I think there's no guarantee that GOOS and GOARCH will comply with the Psiphon format, so it makes sense to always escape.